src/inmanta/plugins.py

```diff
@@ -177,6 +177,18 @@ def get_functions(cls) -> Dict[str, "Type[Plugin]"]:
         """
         return dict(cls.__functions)
 
+    @classmethod
+    def deprecate_function(cls, fnc: Callable, replaced_by: Optional[str] = None) -> None:
+        name = fnc.__name__
+        ns_parts = str(fnc.__module__).split(".")
+        ns_parts.append(name)
+        full_name = "::".join(ns_parts[1:])
+        if full_name in cls.get_functions():
+            cls.__functions[full_name].deprecated = True
+            cls.__functions[full_name].replaced_by = replaced_by
+        else:
+            raise Exception(f"Can not deprecate plugin {full_name} as it does not exist")
```

Comment:
I think this will also be raise if you do 

```
    @plugin
    @deprecated(replaced_by="my_new_plugin")
    def printf():
```

so I don't think this is the best error message.